src/proxy-ui-api/frontend/src/views/AddClient/AddClient.vue

```diff
@@ -1,56 +1,231 @@
+        </v-stepper-content>
+        <!-- Step 2 -->
+        <v-stepper-content step="2">
+          <TokenPage @cancel="cancel" @previous="tokenPrevious" @done="tokenReady" />
+        </v-stepper-content>
+        <!-- Step 3 -->
+        <v-stepper-content step="3">
+          <SignKeyPage @cancel="cancel" @previous="signKeyPrevious" @done="signKeyReady" />
+        </v-stepper-content>
+        <!-- Step 4 -->
+        <v-stepper-content step="4">
+          <WizardPageCsrDetails
+            @cancel="cancel"
+            @previous="csrDetailsPrevious"
+            @done="csrDetailsReady"
+            saveButtonText="action.next"
+          />
+        </v-stepper-content>
+        <!-- Step 5 -->
+        <v-stepper-content step="5">
+          <GenerateCsrPage
+            @cancel="cancel"
+            @previous="generateCsrPrevious"
+            @done="generateCsrReady"
+            saveButtonText="action.next"
+          />
+        </v-stepper-content>
+        <!-- Step 6 -->
+        <v-stepper-content step="6">
+          <FinishPage @cancel="cancel" @previous="finishPrevious" @done="done" />
+        </v-stepper-content>
+      </v-stepper-items>
+    </v-stepper>
+  </div>
 </template>
 
 <script lang="ts">
 import Vue from 'vue';
-import { RouteName } from '@/global';
+import { mapGetters } from 'vuex';
+import HelpIcon from '@/components/ui/HelpIcon.vue';
+import LargeButton from '@/components/ui/LargeButton.vue';
+import SubViewTitle from '@/components/ui/SubViewTitle.vue';
+import ClientDetailsPage from './ClientDetailsPage.vue';
+import TokenPage from './TokenPage.vue';
+import SignKeyPage from './SignKeyPage.vue';
+import FinishPage from './FinishPage.vue';
+import WizardPageCsrDetails from '@/components/wizard/WizardPageCsrDetails.vue';
+import GenerateCsrPage from './GenerateCsrPage.vue';
+
+import { Key, Token } from '@/types';
+import { RouteName, UsageTypes } from '@/global';
+import * as api from '@/util/api';
 
 export default Vue.extend({
-  data: () => ({
-    e1: 0,
-  }),
+  components: {
+    HelpIcon,
+    LargeButton,
+    SubViewTitle,
+    ClientDetailsPage,
+    TokenPage,
+    SignKeyPage,
+    FinishPage,
+    WizardPageCsrDetails,
+    GenerateCsrPage,
+  },
+  props: {},
+  data() {
+    return {
+      currentStep: 1,
+    };
+  },
   methods: {
+    save(): void {
+      this.$store.dispatch('fetchCsrForm').then(
+        (response) => {
+          this.currentStep = 2;
+        },
+        (error) => {
+          this.$store.dispatch('showError', error);
+        },
+      );
+    },
     cancel(): void {
+      this.$store.dispatch('resetState');
+      this.$store.dispatch('resetAddClientState');
+      this.$router.replace({ name: RouteName.Clients });
+    },
+
+    clientDetailsReady(): void {
+      this.currentStep = 2;
+    },
+
+    tokenReady(): void {
+      this.currentStep = 3;
+
+      this.$store.dispatch('fetchLocalMembers').catch((error) => {
```

Comment:
Successful result is handled in the vuex. Error is handled on the "ui side" because it does something visible --> launches the snackbar. Trying to separate the store stuff from ui. Though currently it's more philosophical thing because the snackbar is also called via vuex.